presences/src/main/resources/public/ts/controllers/alerts.ts

```diff
@@ -47,44 +40,55 @@ interface ViewModel {
         type: string
     };
 
-    getStudentAlert(students?: any[], classes?: string[]): void;
+    getStudentAlert(students?: string[], classes?: string[]): Promise<void>;
 
-    searchStudent(value: string): void;
+    searchStudent(value: string): Promise<void>;
 
-    selectStudent(model: any, student: any): void;
+    selectStudent(model: any, student: any): Promise<void>;
 
     searchClass(value: string): Promise<void>;
 
     selectClass(model: any, classObject: any): void;
 
-    dropFilter(object, list): void;
-
-    excludeClassFromFilter(audience): void;
+    dropFilter(object: any, list: any): void;
 
     /*  switch alert type */
-    switchFilter(filter: string): void;
+    switchFilter(filter: string): Promise<void>;
 
     someSelectedAlert(): boolean;
 
-    updateFilter(student?, audience?): void;
-
     selectAll(): void;
 
-    toggleAlert(alert): void;
+    toggleAlert(alert: StudentAlert): void;
 
-    reset(): void;
+    reset(): Promise<void>;
 
     exportAlertCSV(): void;
 }
 
-export const alertsController = ng.controller('AlertsController', ['$scope', '$route', '$location',
-    'SearchService', 'GroupService',
-    function ($scope, $route, $location, SearchService: SearchService, GroupService: GroupService) {
+let that: Controller = null;
```

Comment:
that est utiliser comme singletons dans les méthode on l'on perd l'instance du this

Reply:
Pourquoi ne pas le faire pour chaque méthode qui en ont besoin ?
C'est dommage car sur 15+ méthodes t'en auras surement que 2/3 qui en ont besoin..